solver/src/settlement_submission/retry.rs

```diff
@@ -0,0 +1,136 @@
+use super::{EncodedSettlement, GAS_PRICE_REFRESH_INTERVAL, MAX_GAS};
+use anyhow::Result;
+use contracts::GPv2Settlement;
+use ethcontract::{
+    errors::{ExecutionError, MethodError},
+    jsonrpc::types::Error as RpcError,
+    transaction::{confirm::ConfirmParams, ResolveCondition},
+    web3::error::Error as Web3Error,
+    GasPrice,
+};
+use futures::{stream, Stream, StreamExt};
+use gas_estimation::GasPriceEstimating;
+use primitive_types::U256;
+use std::time::Duration;
+use transaction_retry::{TransactionResult, TransactionSending};
+
+// Create a never ending stream of gas prices based on checking the estimator in fixed intervals
+// and enforcing the minimum increase. Errors are ignored.
+pub fn gas_price_stream(
+    target_confirm_time: Duration,
+    gas_price_cap: f64,
+    estimator: &dyn GasPriceEstimating,
+) -> impl Stream<Item = f64> + '_ {
+    let stream = stream::unfold(true, move |first_call| async move {
+        if !first_call {
+            tokio::time::delay_for(GAS_PRICE_REFRESH_INTERVAL).await;
```

Comment:
ubernit: move constant to this module?

Reply:
Used in several places now so should stay in parent.